infra/azure/modules/auth/main.tf

```diff
@@ -58,12 +58,53 @@ resource "azuread_application_password" "oauth2" {
   application_object_id = azuread_application.oauth2.object_id
 }
 
+resource "random_uuid" "hailctl_oauth2_idenfier_uri_id" {}
+resource "random_uuid" "hailctl_oauth2_scope_id" {}
+
+resource "azuread_application" "hailctl_oauth2" {
+  display_name = "${var.resource_group_name}-hailctl-oauth2"
+
+  identifier_uris = ["api://hail-${random_uuid.hailctl_oauth2_idenfier_uri_id.result}"]
+
+  public_client {
+    redirect_uris = ["http://localhost/"]
+  }
+
+  api {
+    oauth2_permission_scope {
+      admin_consent_description  = "Allow the hail python library to access the Hail Batch service on behalf of the signed-in user."
+      admin_consent_display_name = "hailctl"
+      user_consent_description   = "Allow the hail python library to access the Hail Batch service on your behalf."
```

Comment:
capitalize Python

Reply:
And I guess Hail? I call it the "Hail library" but the package and modules are `hail` obviously.